### PR TITLE
feat: switch to using create table procedure

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Switch to using create table procedure
2. Remove unused code and metrics 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1727

### CODE REVIEW

File: src/meta-srv/src/metasrv/builder.rs [hunk 1]
```diff
@@ -115,6 +117,11 @@ impl MetaSrvBuilder {
         self
     }
 
+    pub fn datanode_clients(mut self, clients: Arc<DatanodeClients>) -> Self {
+        self.datanode_clients = Some(clients);
+        self
+    }
```
Comment (fengjiachun) [inline src/meta-srv/src/metasrv/builder.rs:123]: It seems that it is not necessary to configure datanode_clients, we can create a new one when creating DdlManager.

Comment (WenyXu) [reply src/meta-srv/src/metasrv/builder.rs:123]: > It seems that it is not necessary to configure datanode_clients, we can create a new one when creating DdlManager.

It's used for test purposes, in the IG tests we don't actually connect to a datanode. Instead, we configure a datanode clients.

**Comment by WenyXu** [comment]
PTAL @fengjiachun @MichaelScofield

File: src/meta-srv/src/procedure/create_table.rs
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/create_table.rs:1]: Please add some metrics in create table procedure. At least the removed two in `src/frontend/src/metrics.rs`.

**Review by MichaelScofield — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**